### PR TITLE
fix: properly apply all bindings to service accounts

### PR DESCRIPTION
**Pull request by thefirstofthe300**
All tests are passing:

```
+ kitchen verify ''
-----> Starting Kitchen (v1.23.5)
$$$$$$ Running command `terraform version` in directory /home/seymourd/workspace/github.com/terraform-google-modules/terraform-google-service-accounts
       Terraform v0.11.14
       
       Your version of Terraform is out of date! The latest version
       is 0.12.2. You can update by downloading from www.terraform.io/downloads.html
$$$$$$ Terraform v0.11.14 is supported
WARN: Unresolved or ambigious specs during Gem::Specification.reset:
      tins (~> 1.0)
      Available/installed versions of this gem:
      - 1.20.2
      - 1.19.0
      retriable (>= 2.0, < 4.0)
      Available/installed versions of this gem:
      - 3.1.2
      - 2.1.0
WARN: Clearing out unresolved specs. Try 'gem cleanup <gem>'
Please report a bug if this causes problems.
-----> Setting up <single-service-account-default>...
       Finished setting up <single-service-account-default> (0m0.00s).
-----> Verifying <single-service-account-default>...
$$$$$$ Running command `terraform output -json` in directory /home/seymourd/workspace/github.com/terraform-google-modules/terraform-google-service-accounts/test/fixtures/single_service_account
       {
           "email": {
        "sensitive": false,
        "type": "string",
        "value": "single-account@seymourd-seed.iam.gserviceaccount.com"
           },
           "iam_email": {
        "sensitive": false,
        "type": "string",
        "value": "serviceAccount:single-account@seymourd-seed.iam.gserviceaccount.com"
           },
           "project_id": {
        "sensitive": false,
        "type": "string",
        "value": "seymourd-seed"
           }
       }
Verifying single_service_account gcp

Profile: single_service_account
Version: (not specified)
Target:  gcp://project-factory-30038@seymourd-seed.iam.gserviceaccount.com

  ✔  gcp: GCP Resources
     ✔  Service Account "Terraform-managed service account" project_id should eq "seymourd-seed"
     ✔  Project IAM Binding roles/viewer members should include "serviceAccount:single-account@seymourd-seed.iam.gserviceaccount.com"


Profile: Google Cloud Platform Resource Pack (inspec-gcp)
Version: 0.9.0
Target:  gcp://project-factory-30038@seymourd-seed.iam.gserviceaccount.com

     No tests executed.

Profile Summary: 1 successful control, 0 control failures, 0 controls skipped
Test Summary: 2 successful, 0 failures, 0 skipped
       Finished verifying <single-service-account-default> (0m8.69s).
-----> Setting up <multiple-service-accounts-default>...
       Finished setting up <multiple-service-accounts-default> (0m0.00s).
-----> Verifying <multiple-service-accounts-default>...
$$$$$$ Running command `terraform output -json` in directory /home/seymourd/workspace/github.com/terraform-google-modules/terraform-google-service-accounts/test/fixtures/multiple_service_accounts
       {
           "emails": {
        "sensitive": false,
        "type": "list",
        "value": [
            "test-first@seymourd-seed.iam.gserviceaccount.com",
            "test-second@seymourd-seed.iam.gserviceaccount.com"
        ]
           },
           "project_id": {
        "sensitive": false,
        "type": "string",
        "value": "seymourd-seed"
           }
       }
Verifying multiple_service_accounts gcp

Profile: multiple_service_accounts
Version: (not specified)
Target:  gcp://project-factory-30038@seymourd-seed.iam.gserviceaccount.com

  ✔  gcp: GCP Resources
     ✔  google_service_accounts service_account_emails should include "test-first@seymourd-seed.iam.gserviceaccount.com"
     ✔  Project IAM Binding roles/viewer members should include "serviceAccount:test-first@seymourd-seed.iam.gserviceaccount.com"
     ✔  Project IAM Binding roles/storage.objectViewer members should include "serviceAccount:test-first@seymourd-seed.iam.gserviceaccount.com"
     ✔  google_service_accounts service_account_emails should include "test-second@seymourd-seed.iam.gserviceaccount.com"
     ✔  Project IAM Binding roles/viewer members should include "serviceAccount:test-second@seymourd-seed.iam.gserviceaccount.com"
     ✔  Project IAM Binding roles/storage.objectViewer members should include "serviceAccount:test-second@seymourd-seed.iam.gserviceaccount.com"


Profile: Google Cloud Platform Resource Pack (inspec-gcp)
Version: 0.9.0
Target:  gcp://project-factory-30038@seymourd-seed.iam.gserviceaccount.com

     No tests executed.

Profile Summary: 1 successful control, 0 control failures, 0 controls skipped
Test Summary: 6 successful, 0 failures, 0 skipped
       Finished verifying <multiple-service-accounts-default> (0m5.26s).
-----> Kitchen is finished. (0m16.85s)
```

### CODE REVIEW

**Comment by ludoo** [comment]
Hey, actually this won't work as the count is the product of the number of roles, and the number of service accounts. The fact that test pass is only a flag that tests don't exercise this point properly. :)

Let me try to explain. Imagine you have three roles, and four service accounts. The count is 12. The index for the project/role list needs cycle between 0 and 2. The index for the service account list needs to cycle between 0 and 3. Your patch makes the index for the project/role list cycle between 0 and 11, which is clearly incorrect.

**Comment by thefirstofthe300** [comment]
You're right (obviously). I didn't quite think this one all the way through. I'm chasing down a bug for sure, but the bug only triggers when the number of roles to be applied to the service account exceeds the number of service accounts that are created.

Think of this scenario, if you have a single service account and need to apply four roles to it, the value of 1 % {1-4} is always 1 so the cycle you describe never actually occurs. The first permission in the list of provided permissions is applied four different times. Same goes for if you have a two service accounts and need to apply four roles to it. `{1-2} % {1-4}` cycles between 1 and 2.

To sum up the bug, you can only apply a list of permissions less than or equal to the number of service accounts that are created.

**Comment by ludoo** [comment]
Hmmm I think I know what's the issue. I can give it a shot between later today and tomorrow, unless you want to give it a go yourself. And thanks for spotting this ofc.

**Comment by thefirstofthe300** [comment]
I'm under a time crunch and have a lot of other things to get done so if
you think you can get it worked out quickly, I'd welcome the fix. :)

**Comment by ludoo** [comment]
So, I wrote a simple Python script to test combinations, as I'm too dumb to look at the mathematics behind this. ;) It basically does `count index % projects` for project index, and `count index / projects` for sa index.

Your case:
```
num projects = 4, num service accounts = 1

count project/role service account
----- ------------ ---------------
    0            0               0
    1            1               0
    2            2               0
    3            3               0
```

Even numbers:
```
num projects = 2, num service accounts = 2

count project/role service account
----- ------------ ---------------
    0            0               0
    1            1               0
    2            0               1
    3            1               1
```

Odd numbers:
```
num projects = 3, num service accounts = 3

count project/role service account
----- ------------ ---------------
    0            0               0
    1            1               0
    2            2               0
    3            0               1
    4            1               1
    5            2               1
    6            0               2
    7            1               2
    8            2               2
```

Even/odd combinations:
```
num projects = 3, num service accounts = 2

count project/role service account
----- ------------ ---------------
    0            0               0
    1            1               0
    2            2               0
    3            0               1
    4            1               1
    5            2               1

num projects = 2, num service accounts = 3

count project/role service account
----- ------------ ---------------
    0            0               0
    1            1               0
    2            0               1
    3            1               1
    4            0               2
    5            1               2
```

I don't see how this is not working. ;)
